jgrapht-core/src/test/java/org/jgrapht/perf/spanning/MinimumSpanningTreePerformanceTest.java

```diff
@@ -0,0 +1,192 @@
+public class MinimumSpanningTreePerformanceTest
+    extends TestCase
+{
+    private static final int PERF_BENCHMARK_VERTICES_COUNT = 1000;
+    private static final double PERF_BENCHMARK_EDGES_PROP = 0.5;
+    private static final int WARMUP_REPEAT = 5;
+    private static final int REPEAT = 10;
+    private static final long SEED = 13l;
+
+    private static abstract class BenchmarkBase
+    {
+        protected Random rng = new Random(SEED);
+        protected GraphGenerator<Integer, DefaultWeightedEdge, Integer> generator = null;
+        protected DirectedGraph<Integer, DefaultWeightedEdge> graph;
+
+        abstract MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph);
+
+        public void setup()
+        {
+            if (generator == null) {
+                // lazily construct generator
+                generator = new GnpRandomGraphGenerator<>(
+                    PERF_BENCHMARK_VERTICES_COUNT, PERF_BENCHMARK_EDGES_PROP, rng, false);
+            }
+
+            DirectedWeightedPseudograph<Integer, DefaultWeightedEdge> weightedGraph =
+                new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+            this.graph = weightedGraph;
+
+            generator.generateGraph(graph, new IntegerVertexFactory(), null);
+
+            for (DefaultWeightedEdge e : weightedGraph.edgeSet()) {
+                weightedGraph.setEdgeWeight(e, rng.nextDouble());
+            }
+        }
+
+        public void run()
+        {
+            MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> algo = createSolver(graph);
+            algo.getSpanningTree();
+        }
+    }
+
+    public static class PrimBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new PrimMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Prim";
+        }
+    }
+
+    public static class KruskalBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new KruskalMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Kruskal";
+        }
+    }
+
+    public static class BoruvkaBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new BoruvkaMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Boruvka";
+        }
+    }
+
+    public void testBenchmark()
```

Comment:
Is there a reason why you used the Stopwatch class instead of the build-in mechanisms of jmh like you did for the PathGrowingWeightedMatchingPerformanceTest class? Personally I would say this approach is fine, but I'm not an expert in building proper performance tests.